src/main/java/htsjdk/samtools/BinningIndexBuilder.java

```diff
@@ -131,9 +137,11 @@ public void processFeature(final FeatureToBeIndexed feature) {
             largestIndexSeen = endWindow;
         }
 
-        // set linear index at every 16K window that this feature overlaps
+        // Set the linear index at every 16K window that this feature overlaps, but only if this chunk
+        // start is strictly earlier than any previously seen chunk start for the window, or if the window
+        // is uninitialized (this is the first feature overlapping this window).
         for (int win = startWindow; win <= endWindow; win++) {
-            if (index[win] == 0 || chunkStart < index[win]) {
+            if (index[win] == UNINITIALIZED_WINDOW || chunkStart < index[win]) {
```

Comment:
Add comment that this needs to be this way to deal with records that legitimately start at position 0